README.rdoc

```diff
@@ -123,6 +123,70 @@ is raised if the job is killed.
 
   at(steps_completed, total_steps, "${steps_completed} of #{total_steps} steps completed!")
 
+=== Distributed job
+
+With Status you can create a destributed parent/child job, that will run some part of the processing,
+in multiple, potentially parallel child jobs and then do somthing when all child jobs will finish.
+To do this you first need to define <tt>perform_child</tt> method. This method will be called in child jobs
+instead of <tt>perform</tt>. Then, in the parent's <tt>perform</tt>, you need initiate parent using <tt>init_parent(total)</tt>,
+and finally queue children using <tt>enqueue_child(options)</tt>.
+The progress is tracked automatically using regular "num" and "total" fields. The "total" will be set to
+the "total" from <tt>init_parent</tt> and <tt>num</tt> will be incresed by 1 for each finished job, so DO NOT change them manually.
+When all child jobs will finish, the last one will call <tt>onsuccess</tt> in its own context.
+
+Example:
+
+    class VeryHeavyJob
+      include Resque::Plugins::Status
+
+      def perform
+        init_parent(options['items'].size)
```

Comment:
As for me `Batch` terminology sounds better than parent/child. Honestly say you even don't need to start the parent `perform` method to create children.  You can create them from the application and combine them under a batch. I like this approach https://github.com/annkissam/resque-batch
```
batch = Resque::Plugins::Batch.new()
batch.enqueue(Job, 11)
batch.enqueue(Job, 12, "test2")
result = batch.perform
```
This approach allows you to run different types of jobs in parallel. For example, you need to run 10 `HistoricalPriceLoader` jobs and 1 `GetFundFundamentals`  and only after that run `GraderJob`.
You can also take a look onto https://github.com/drfeelngood/resque-batched-job

Reply:
Here I preferred to have parent/child specifically because this is a pattern for creating a single distributed job, not a batch of unrelated jobs. I don't see a point of having some other place in code to think about batching. Pattern that would leverage this is mainly resque-scheduled job that will do some initial calculation and then split itself into many and in the end aggregate results or queue next job(s).